File: app/src/main/java/com/example/mvp_sample/MainActivity.java
package com.example.mvp_sample;

import android.content.Intent;
import android.view.View;

import com.example.mvp_sample.base.BaseActivity;
import com.example.mvp_sample.mvc.ArticlesActivity;
import com.example.mvp_sample.mvp_activity.ArticlesMvpActivity;
import com.example.mvp_sample.mvp_fragment.FragmentWithActivity;

public class MainActivity extends BaseActivity {

    @Override
    protected int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    protected void initViews() {

    }

    @Override
    protected void initDatas() {

    }

    public void mvcClick(View view) {
        startActivity(new Intent(mContext, ArticlesActivity.class));
    }

    public void mvpActivityClick(View view) {
        startActivity(new Intent(mContext, ArticlesMvpActivity.class));
    }

    public void mvpFragmentClick(View view) {
        startActivity(new Intent(mContext, FragmentWithActivity.class));
    }
}

File: mvvm_clean_hilt/src/main/java/com/example/mvvm_clean_hilt/ui/MainActivity.kt
package com.example.mvvm_coroutines_retrofit_flow_hilt.ui

import android.content.Intent
import android.os.Bundle
import android.view.View
import com.example.common.base.BaseActivity
import com.example.mvvm_coroutines_retrofit_flow_hilt.databinding.ActivityMainBinding
import com.example.mvvm_coroutines_retrofit_flow_hilt.ui.articlelist.ArticleListActivity
import com.example.mvvm_coroutines_retrofit_flow_hilt.ui.login.LoginActivity
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.update

class MainActivity : BaseActivity<ActivityMainBinding>() {

    override fun createViewBinding(): ActivityMainBinding {
        return ActivityMainBinding.inflate(layoutInflater)
    }

    override fun initViews() {
        mViewBinding.root.post { }
    }

    override fun initData(savedInstanceState: Bundle?) {

    }

    fun toLoginActivity(view: View) {
        startActivity(Intent(this, LoginActivity::class.java))
    }

    fun toArticleListActivity(view: View) {
        startActivity(Intent(this, ArticleListActivity::class.java))
    }
}